.changesets/feat_geal_cache_size_metric.md

```diff
@@ -0,0 +1,5 @@
+### Add a metric to track the cache size ([Issue #2522](https://github.com/apollographql/router/issues/2522))
+
+The new `apollo_router_cache_size` metric will indicate the current size of in memory caches.
+
```

Comment:
For future self, this is a **gauge** of the length/**count** of the cache entries.